app/master/cluster_master.py

```diff
@@ -228,7 +224,22 @@ def handle_request_for_new_build(self, build_params):
         if build_request.is_valid():
             build = Build(build_request)
             self._all_builds_by_id[build.build_id()] = build
-            self._request_queue.put(build)
+
```

Comment:
It appears to me that the project_type can be initialized during construction or initialized lazily since we have all the necessary information when we instantiate it.